### PR TITLE
Add recent clan chats to clan chat plugin

### PR DESCRIPTION
**Pull request by raiyni**
Closes https://github.com/runelite/runelite/issues/4833

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java [hunk 1]
```diff
@@ -52,11 +63,46 @@
...
-		period = 600,
-		unit = ChronoUnit.MILLIS
-	)
-	public void updateClanChatTitle()
+	@Inject
+	private ClanChatConfig config;
+
+	private List<String> chats = new ArrayList<>();
+
+	private static final int MAX_CHATS = 10;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java]: Constants should be at top of the file

File: runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java [hunk 2]
```diff
@@ -52,11 +63,46 @@
...
+	{
+		return configManager.getConfig(ClanChatConfig.class);
+	}
+
+	@Override
+	public void startUp()
+	{
+		if (!config.chatsData().isEmpty())
+		{
+			chats = Arrays.stream(config.chatsData().split(",")).collect(Collectors.toList());
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java]: Splitter.on(",").splitToList() ?

Comment (raiyni) [reply runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java]: Splitter.On() creates an immutable list.

Comment (raiyni) [reply runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java]: I see you remedied that.

**Comment by Gamer1120** [comment]
A delete button after each recent chat would be a nice to have. If you make a typo and join the wrong CC, you wouldn't want it to appear on this list I think. Maybe outside of the scope of this PR and maybe unnecessary.

**Comment by deathbeam** [comment]
Probably not, because this is keeping history of recently joined clan chats and is not your favorited/liked clan chats or something like that

**Comment by deathbeam** [comment]
Needs rebase

**Pull request merged**